controllers/volumereplicationgroup_controller.go

```diff
@@ -1718,22 +1744,24 @@ func getStatusStateFromSpecState(state ramendrv1alpha1.ReplicationState) ramendr
 //    VRG.conditions.Available.Status = false
 //    VRG.conditions.Available.Reason = Progressing
 //
-func (v *VRGInstance) updateVRGConditions() {
+func (v *VRGInstance) updateVRGDataReadyCondition() {
 	vrgReady := true
 	vrgProgressing := false
 
 	for _, protectedPVC := range v.instance.Status.ProtectedPVCs {
 		dataReadyCondition := findCondition(protectedPVC.Conditions, VRGConditionTypeDataReady)
 		if dataReadyCondition == nil {
 			vrgReady = false
-
+			// When will we hit this condition? If it is due to a race condition,
+			// why treat it as an error instead of progressing?
 			break
 		}
 
 		if dataReadyCondition.Reason == VRGConditionReasonProgressing {
 			vrgReady = false
 			vrgProgressing = true
-
+			// Breaking out in this case may be incorrect, as another PVC could
+			// have a more serious `error` condition, isn't it?
```

Comment:
@raghavendrabhat Please take a look at [updateVRGClusterDataProtectedCondition](https://github.com/RamenDR/ramen/pull/255/files#diff-428a43c23cf418d4022f864ffab5269355b5e7521d3fb9e5f7bdf6b202b184d8R1813) to see my take on the answers to the two questions I've raised in this method.

Reply:
Agree. We should check all the PVCs to be able to correctly report the status of VRG. Like you said, it is possible that some other PVC may be in error condition and it would be more appropriate to report error instead of progressing for VRG.